src/libpriv/rpmostree-scripts.c

```diff
@@ -259,19 +276,43 @@ run_script_in_bwrap_container (int rootfs_fd,
   if (!bwrap)
     goto out;
 
-  if (stdin_fd >= 0)
-    rpmostree_bwrap_set_child_setup (bwrap, script_child_setup_stdin, GINT_TO_POINTER (stdin_fd));
+
+  struct ChildSetupData data = { .stdin_fd = stdin_fd };
+
+  const char *id = glnx_strjoina ("rpm-ostree(", pkg_script, ")");
+  data.stdout_fd = stdout_fd = sd_journal_stream_fd (id, LOG_INFO, 0);
+  if (stdout_fd < 0)
+    {
+      glnx_throw_errno_prefix (error, "While creating stdout stream fd");
+      goto out;
+    }
+
+  data.stderr_fd = stderr_fd = sd_journal_stream_fd (id, LOG_ERR, 0);
```

Comment:
Does systemd actually use separate streams for this by default?  What you're doing makes a ton of sense, I like it, I just wonder why systemd doesn't do this by default (it looks like the service default is `StandardOutput=journal` and `StandardError=inherit`, which results in a pure dup).

Reply:
Yeah, I'm not sure why that wasn't the default behaviour to begin with, though it looks like there's work pending to change this: https://github.com/systemd/systemd/issues/5019, https://github.com/systemd/systemd/pull/6599.

Reply:
Ah hah nice work on finding that.